### PR TITLE
SUBMARINE-241. Workbench web ng

### PR DESCRIPTION
**Pull request by lleohao**
### What is this PR for?
Develop Web Framework with Angular and Ng-zero
Init project and add user module ( login and regisiter )

### What type of PR is it?
[Feature]

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-241

### How should this be tested?
No test

### Screenshots (if appropriate)
image

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by jiwq** [comment]
Thanks @lleohao contributing. Before we create a PR, we need create the corresponding JIRA. More info see https://github.com/apache/hadoop-submarine/tree/master/docs/community

File: submarine-workbench/workbench-web-ng/src/app/services/local-storage.service.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+  }
+
+  set(key: string, value: string | number | object | any[]) {
+    try {
+      const saveValue = JSON.stringify(value);
+
+      window.localStorage.setItem(this.generateKey(key), saveValue);
+    } catch (e) {
+      // empty
```
Comment (liuxunorg) [inline submarine-workbench/workbench-web-ng/src/app/services/local-storage.service.ts]: Don't need to print the log?？

File: submarine-workbench/workbench-web-ng/src/assets/background.svg [hunk 2]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg width="1361px" height="609px" viewBox="0 0 1361 609" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <!-- Generator: Sketch 46.2 (44496) - http://www.bohemiancoding.com/sketch -->
+    <title>Group 21</title>
+    <desc>Created with Sketch.</desc>
+    <defs></defs>
+    <g id="Ant-Design-Pro-3.0" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <g id="账户密码登录-校验" transform="translate(-79.000000, -82.000000)">
```
Comment (liuxunorg) [inline submarine-workbench/workbench-web-ng/src/assets/background.svg]: Please change to English.

File: submarine-workbench/workbench-web-ng/src/assets/submarine-logo.svg [hunk 3]
```diff
@@ -0,0 +1 @@
....42,0,0,0-47,9.58h0L208.13,632a2,2,0,0,0,0,3.53l100.59,43.65,2.67,1.14a52,52,0,0,0,18.15,3.44h27.32a20,20,0,0,0,19.67-16.44h0a31.32,31.32,0,0,1,17.58-22.91h0l5.24-2c12.41-4.58,22.37-11.82,27.22-23.25l.4-1a25.31,25.31,0,0,0-23.63-34.38Z"/><g class="cls-13"><path class="cls-14" d="M464.7,582.73H342.37v0a119.86,119.86,0,1,0,118,88.08h4.28Z"/><path class="cls-15" d="M444.67,603.08V583.75H390.22a86.42,86.42,0,1,0,54.45,19.33Z"/><path class="cls-1" d="M403.33,583.75H344.16a121.42,121.42,0,0,0-47,9.58h0L208.13,632a2,2,0,0,0,0,3.53l100.59,43.65,2.67,1.14a52,52,0,0,0,18.15,3.44h27.32a20,20,0,0,0,19.67-16.44h0a31.32,31.32,0,0,1,17.58-22.91h0l5.24-2c12.41-4.58,22.37-11.82,27.22-23.25l.4-1a25.31,25.31,0,0,0-23.63-34.38Z"/><path class="cls-16" d="M409.18,637.84a28.73,28.73,0,0,1-12.9,3.61h0a62.88,62.88,0,0,0-55.84,42.3h1.69a63.3,63.3,0,1,0,123.06,20.88C465.18,669.68,406.79,639.29,409.18,637.84Z"/><path class="cls-1" d="M403.33,583.75H344.16a121.42,121.42,0,0,0-47,9.58h0L208.13,632a2,2,0,0,0,0,3.53l100.59,43.65,2.67,1.14a52,52,0,0,0,18.15,3.44h27.32a20,20,0,0,0,19.67-16.44h0a31.32,31.32,0,0,1,17.58-22.91h0l5.24-2c12.41-4.58,22.37-11.82,27.22-23.25l.4-1a25.31,25.31,0,0,0-23.63-34.38Z"/></g></svg>
```
Comment (liuxunorg) [inline submarine-workbench/workbench-web-ng/src/assets/submarine-logo.svg]: Please change to English.

**Comment by liuxunorg** [comment]
@lleohao , You need change PR title to `SUBMARINE-241. Workbench web ng`.

**Comment by lleohao** [comment]
@liuxunorg please review

File: submarine-workbench/workbench-web-ng/src/app/pages/user/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      password: [null, [Validators.required]],
+      remember: [true]
+    });
+  }
+
+  loginSuccess() {
+    this.router.navigate(['/manager/user']);
+
+    setTimeout(() => {
+      this.nzNotificationService.success('欢迎', '欢迎回来');
```
Comment (liuxunorg) [inline submarine-workbench/workbench-web-ng/src/app/pages/user/login/login.component.ts]: Change to English.

File: submarine-workbench/workbench-web-ng/src/app/pages/user/login/login.component.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    }, 1000);
+  }
+
+  requestFailed(error: Error) {
+    this.nzNotificationService.error('请求错误', error.message || '请求出现错误，请稍后再试', {
```
Comment (liuxunorg) [inline submarine-workbench/workbench-web-ng/src/app/pages/user/login/login.component.ts]: Change to English.

**Comment by liuxunorg** [comment]
LGTM
@lleohao , Thank you contribute.

**Comment by yuanzac** [comment]
Thanks @lleohao for the contribution and @liuxunorg @jiwq for the review.

**Pull request closed without merging**